__tests__/app.test.js

```diff
@@ -0,0 +1,198 @@
+        expect(typeof body.article.author).toBe("string");
+        expect(typeof body.article.title).toBe("string");
+        expect(typeof body.article.article_id).toBe("number");
+        expect(typeof body.article.body).toBe("string");
+        expect(typeof body.article.topic).toBe("string");
+        expect(new Date(body.article.created_at)).toBeInstanceOf(Date);
+        expect(typeof body.article.votes).toBe("number");
+        expect(typeof body.article.article_img_url).toBe("string");
+      });
+  });
+
+  it("GET:404 responds with 'not found' if provided with a valid article_id but doesn't exist", () => {
+    return request(app)
+      .get("/api/articles/14")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+
+  it("GET:400 responds with 'bad request' if article if provided with an invalid article_id", () => {
+    return request(app)
+      .get("/api/articles/chickens")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+});
+
+describe("/api/articles", () => {
+  it("GET:200 responds with an array of the correct length", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        expect(Array.isArray(body.articles)).toBe(true);
+        expect(body.articles).toHaveLength(13)
+      });
+  });
+
+  it("GET:200 responds with articles with the correct properties", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        body.articles.forEach((article) => {
+          const requestedKeys = [
+            "author",
+            "title",
+            "article_id",
+            "topic",
+            "created_at",
+            "votes",
+            "article_img_url",
+          ];
+
+          expect(article).toContainKeys(requestedKeys);
+
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.article_id).toBe("number");
+          expect(typeof article.topic).toBe("string");
+          expect(new Date(article.created_at)).toBeInstanceOf(Date);
+          expect(typeof article.votes).toBe("number");
+          expect(typeof article.article_img_url).toBe("string");
+        });
+      });
+  });
+
+  it('GET:200 does not return the "body" key for any items ', () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        body.articles.forEach((article) => {
+          expect(article).not.toContainKey("body");
+        });
+      });
+  });
+
+  it("GET:200 sorts the returned articles in descending order by date", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+});
```

Comment:
It is up to you, but you could combine these three tests into one. This is ok in integration testing. Checking the order and the value of the response in one.